packages/react-query/src/shared/hooks/createHooksInternal.tsx

```diff
@@ -503,26 +503,6 @@ export function createRootHooks<
     useSuspenseInfiniteQuery,
   };
 }
-/* istanbul ignore next */
-/**
- * Hack to infer the type of `createReactQueryHooks`
- * @link https://stackoverflow.com/a/59072991
- */
-class GnClass<TRouter extends AnyRouter, TSSRContext = unknown> {
-  fn() {
-    return createRootHooks<TRouter, TSSRContext>();
-  }
-}
-
-type returnTypeInferer<TType> = TType extends (
-  a: Record<string, string>,
-) => infer U
-  ? U
-  : never;
-type fooType<TRouter extends AnyRouter, TSSRContext = unknown> = GnClass<
-  TRouter,
-  TSSRContext
->['fn'];
```

Comment:
Unrelated but is nice to get rid of this, our TS req is high enough not to require this